ballerina/utils.bal

```diff
@@ -0,0 +1,216 @@
+}
+
+# Serialize the array of records according to the form style.
+#
+# + parent - Parent record name
+# + value - Array of records to be serialized
+# + style - Defines how multiple values are delimited
+# + explode - Specifies whether arrays and objects should generate separate parameters
+# + return - Serialized record as a string
+isolated function getSerializedRecordArray(string parent, record {}[] value, string style = FORM, boolean explode = true) returns string {
+    string[] serializedArray = [];
+    if style == DEEPOBJECT {
+        int arayIndex = 0;
+        foreach var recordItem in value {
+            serializedArray.push(getDeepObjectStyleRequest(parent + "[" + arayIndex.toString() + "]", recordItem), "&");
+            arayIndex = arayIndex + 1;
+        }
+    } else {
+        if !explode {
+            serializedArray.push(parent, "=");
+        }
+        foreach var recordItem in value {
+            serializedArray.push(getFormStyleRequest(parent, recordItem, explode), ",");
+        }
+    }
+    _ = serializedArray.pop();
+    return string:'join("", ...serializedArray);
+}
+
+# Get Encoded URI for a given value.
+#
+# + value - Value to be encoded
+# + return - Encoded string
+isolated function getEncodedUri(anydata value) returns string {
+    string|error encoded = url:encode(value.toString(), "UTF8");
+    if encoded is string {
+        return encoded;
+    }
+    return value.toString();
+}
+
+# Generate query path with query parameter.
+#
+# + queryParam - Query parameter map
+# + encodingMap - Details on serialization mechanism
+# + return - Returns generated Path or error at failure of client initialization
+isolated function getPathForQueryParam(map<anydata> queryParam, map<Encoding> encodingMap = {}) returns string|error {
+    string[] param = [];
+    if queryParam.length() > 0 {
+        param.push("?");
+        foreach var [key, value] in queryParam.entries() {
+            if value is () {
+                _ = queryParam.remove(key);
+                continue;
+            }
+            Encoding encodingData = encodingMap.hasKey(key) ? encodingMap.get(key) : defaultEncoding;
+            if value is SimpleBasicType {
+                param.push(key, "=", getEncodedUri(value.toString()));
+            } else if value is SimpleBasicType[] {
+                param.push(getSerializedArray(key, value, encodingData.style, encodingData.explode));
+            } else if value is record {} {
+                if encodingData.style == DEEPOBJECT {
+                    param.push(getDeepObjectStyleRequest(key, value));
+                } else {
+                    param.push(getFormStyleRequest(key, value, encodingData.explode));
+                }
+            } else {
+                param.push(key, "=", value.toString());
+            }
+            param.push("&");
+        }
+        _ = param.pop();
+    }
+    string restOfPath = string:'join("", ...param);
+    return restOfPath;
```

Comment:
```suggestion
    return string:'join("", ...param);
```

Reply:
Fixed in [0971b94](https://github.com/ballerina-platform/module-ballerinax-docusign.dsclick/pull/1/commits/0971b94087e563046449c53e2711d3e7c512e6e2)